FactoryMethod/src/main/java/com/ben/notification/FactoryPatternUML.puml

```diff
@@ -0,0 +1,29 @@
+@startuml
+header AbstractFactoryPattern
+
+interface Notification{
+    String notifyUser()
+}
+class EmailNotification{
+    String notifyUser()
+} 
+class TextNotification{
+    String notifyUser()
+} 
+EmailNotification ..> Notification
+TextNotification ..> Notification
+
+class NotificationFactory{
+    Notification createNotification()
+    new TextNotification()
```

Comment:
`new TextNotification()`/`new EmailNotification()` are implementation-specific here, so they shouldn't be included in your UML diagram. You should only include methods/fields on the class itself